# **Exercise: Python Lists**

1.Let us say your expense for every month are listed below,

January - 2200

February - 2350

March - 2600

April - 2130

May - 2190

Create a list to store these monthly expenses and using that find out,

1. In Feb, how many dollars you spent extra compare to January?
2. Find out your total expense in first quarter (first three months) of the year.
3. Find out if you spent exactly 2000 dollars in any month
4. June month just finished and your expense is 1980 dollar. Add this item to our monthly expense list
5. You returned an item that you bought in a month of April and
got a refund of 200$. Make a correction to your monthly expense list
based on this

In [2]:
expenses = [2200, 2350, 2600, 2130, 2190]
#how many dollors you spent extra compare to january?
print("the dollor spent extra compare to january is:",expenses[1]-expenses[0])
#Find out your total expense in first quarter (first three months) of the year.
print("the total expense in first quarter is",expenses[0]+expenses[1]+expenses[2])
#Find out if you spent exactly 2000 dollars in any month
print("Did I spend 2000$ in any month?",2000 in expenses)
#June month just finished and your expense is 1980 dollar. Add this item to our monthly expense list
expenses.append(1980)
print("the june expense is:",expenses)
#you returned an item that you bought in a month of April and got a refund of 200$. Make a correction to your monthly expense list based on this
expenses[3]=expenses[3]-200
print("the april expense is:",expenses)


the dollor spent extra compare to january is: 150
the total expense in first quarter is 7150
Did I spend 2000$ in any month? False
the june expense is: [2200, 2350, 2600, 2130, 2190, 1980]
the april expense is: [2200, 2350, 2600, 1930, 2190, 1980]


**Answer:**Extra spent in February compared to January: 150

Total expense in first quarter: 7150

Did I spend exactly $2000 in any month? False

Expenses after adding June: [2200, 2350, 2600, 2130, 2190, 1980]

Expenses after April refund: [2200, 2350, 2600, 1930, 2190, 1980]

**explanation:**
expenses = [2200, 2350, 2600, 2130, 2190]

stores monthly expenses from Jan to May.


expenses[1] - expenses[0] compares February to January.


sum(expenses[0:3]) sums up Jan, Feb, and Mar.


2000 in expenses checks if 2000 is present in the list.


expenses.append(1980) adds June's expense.


expenses[3] = expenses[3] - 200 subtracts $200 refund from April's expense.

2.You have a list of your favourite marvel super heros.

In [ ]:
heros=['spider man','thor','hulk','iron man','captain america']

Using this find out,

In [ ]:
1. Length of the list
2. Add 'black panther' at the end of this list
3. You realize that you need to add 'black panther' after 'hulk',
   so remove it from the list first and then add it after 'hulk'
4. Now you don't like thor and hulk because they get angry easily :)
   So you want to remove thor and hulk from list and replace them with doctor strange (because he is cool).
   Do that with one line of code.
5. Sort the heros list in alphabetical order (Hint. Use dir() functions to list down all functions available in list)

In [9]:
heros=['spider man','thor','hulk','iron man','captain america']
#Length of the list
len(heros)


5

**Explanation:**

The list heros contains 5 elements.

len(heros) returns the total number of items in the list.

In [10]:
#Add 'black panther' at the end of this list
heros.append("black panther")
print(heros)

['spider man', 'thor', 'hulk', 'iron man', 'captain america', 'black panther']


**Explanation:**

append("black panther") adds the new hero to the end of the list.

The list now contains 6 heroes including "black panther" at the last position.

In [13]:
#You realize that you need to add 'black panther' after 'hulk',so remove it from the list first and then add it after 'hulk'
heros = ['spider man', 'thor', 'hulk', 'iron man', 'captain america', 'black panther']
heros.remove("black panther")
heros.insert(3,"black panther")
print(heros)

['spider man', 'thor', 'hulk', 'black panther', 'iron man', 'captain america']


**Explanation:**

'hulk' is at index 2, so inserting "black panther" at index 3 places it right after hulk.

First, remove() deletes "black panther" from the end.

Then insert(3, "black panther") places it correctly.

In [14]:
#Now you don't like thor and hulk because they get angry easily :)So you want to remove thor and hulk from list and replace them with doctor strange (because he is cool). Do that with one line of code.
heros = ['spider man', 'thor', 'hulk', 'iron man', 'captain america', 'black panther']
heros[1:3] = ['doctor strange']
print(heros)

['spider man', 'doctor strange', 'iron man', 'captain america', 'black panther']


**Explanation:**

heros[1:3] selects the items at index 1 and 2 ('thor' and 'hulk').

Replacing that slice with ['doctor strange'] removes both and adds one in their place.

Clean, short, and effective!

In [18]:
#Sort the heros list in alphabetical order (Hint. Use dir() functions to list down all functions available in list)
heros = ['spider man', 'thor', 'hulk', 'iron man', 'captain america', 'black panther']
heros.sort()
print(heros)

['black panther', 'captain america', 'hulk', 'iron man', 'spider man', 'thor']


**Explanation:**

.sort() is an in-place method that sorts the list in alphabetical order.

You can use dir(list) to explore all available methods for list objects, including append, insert, remove, sort, etc.